app/javascript/react/components/Graph/MeasurementComponent.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+import {
+  selectFixedExtremes,
+  selectFixedStreamShortInfo,
+} from "../../store/fixedStreamSelectors";
+import { selectThresholds } from "../../store/thresholdSlice";
+import { getColorForValue } from "../../utils/thresholdColors";
+import * as S from "../Modals/SessionDetailsModal/SessionDetailsModal.style";
+import { MeasurementContainer } from "./Graph.style";
+
+const MeasurementComponent: React.FC = () => {
+  const { t } = useTranslation();
+
+  const extremes = useSelector(selectFixedExtremes);
+  const thresholds = useSelector(selectThresholds);
+  const { unitSymbol } = useSelector(selectFixedStreamShortInfo);
+
+  const { minMeasurementValue, maxMeasurementValue, averageValue } = extremes;
+
+  return (
+    <MeasurementContainer>
+      <S.AverageValueContainer>
+        <S.AverageDot $color={getColorForValue(thresholds, averageValue)} />
```

Comment:
I'd optimize these calculations using memos - by memoizing the result, you avoid recalculating the color values on every render unless the dependencies change 

```
const averageValueColor = useMemo(() => getColorForValue(thresholds, averageValue), [thresholds, averageValue]);
const minMeasurementValueColor = useMemo(() => getColorForValue(thresholds, minMeasurementValue), [thresholds, minMeasurementValue]);
const maxMeasurementValueColor = useMemo(() => getColorForValue(thresholds, maxMeasurementValue), [thresholds, maxMeasurementValue]);
```